aiohttp/client.py

```diff
@@ -271,7 +271,22 @@ def __init__(
         if timeout is sentinel or timeout is None:
             self._timeout = DEFAULT_TIMEOUT
         else:
-            self._timeout = timeout
+            if not isinstance(timeout, (ClientTimeout, int, float)):
+                raise ValueError(
+                    f"timeout parameter cannot be of {type(timeout)} type, "
+                    "please use 'timeout=ClientTimeout(...)'",
+                )
+            if not isinstance(timeout, ClientTimeout):
+                self._timeout = ClientTimeout(total=timeout)
+                warnings.warn(
+                    "parameter 'timeout' of type 'float' or 'int' "
+                    "is deprecated, please use "
+                    "'timeout=ClientTimeout(total=...)'",
+                    DeprecationWarning,
+                    stacklevel=2,
+                )
```

Comment:
I don't think we need a deprecation warning if it never worked before. You appear to be adding a feature and deprecating it at the same time.

Reply:
Since this test is still here I'm afraid to break someone's code, yes it fails int `.get()` later, but might not be the case for other methods
https://github.com/aio-libs/aiohttp/blob/10ecad886fac40722f654b983a95d49e77f657ea/tests/test_client_session.py#L782-L785